### PR TITLE
Feature/registration

### PR DESCRIPTION
**Pull request by erolcanozgen**

### CODE REVIEW

File: backend/app/controllers/loginController.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+exports.loginWithGoogle = (req, res) => {
+    // Validate request
+    if (!req.body.id) {
+        res.status(400).send({
+            message: "Content can not be empty!"
+        });
+        return;
+    }
+    db.google_account.findByPk(req.body.id, {
+        include: [
```
Comment (ibrahimseckin) [inline backend/app/controllers/loginController.js:12]: what is the reason of this include?

Comment (ibrahimseckin) [reply backend/app/controllers/loginController.js:12]: Are you using that user during the account creation?

File: backend/app/controllers/loginController.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }).then((user) => {
+        db.twitter_account.create({
+            twitter_id: twitterAccount.id,
+            userId: user.id
+        }).then(() => {
+            res.send(true)
+        }).catch((err) => {
+            res.status(500).send({
+                message:
+                    err.message || "error occurred while creating the google account."
```
Comment (ibrahimseckin) [inline backend/app/controllers/loginController.js:88]: typo "twitter" instead of "google"

File: backend/lib/auth.controller.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 exports.google = (req, res) => {
     const io = req.app.get('io')
+    console.log(req.user)
```
Comment (ibrahimseckin) [inline backend/lib/auth.controller.js:3]: Will we need it? Otherwise, can you remove the log?

File: backend/server.js [hunk 4]
```diff
@@ -42,9 +43,12 @@ const io = socketio(server)
 app.set('io', io)
 
 
+
 app.use(express.static(path.join(__dirname, '/../client/build')));
 // Direct other requests to the auth router
-app.use('/', authRouter)
+app.use('/', authRouter);
+
+routes.loginRoutes(app);
```
Comment (ibrahimseckin) [inline backend/server.js:51]: What does it mean?

File: client/src/components/login/components/registration/registration.js [hunk 5]
```diff
@@ -2,33 +2,62 @@ import React, { Component } from 'react'
...
-              <button type="submit" class="btn btn-primary btn-block"> Create Account  </button>
-           </div>
+          <div className="form-group input-group">
+            <div className="input-group-prepend">
+              <span className="input-group-text"> <FontAwesomeIcon icon={faUser} /></span>
+            </div>
+            <input className="form-control" placeholder="Full name" type="text" value={this.state.user.name} onChange={(e) => { this.handleChange(e) }} />
+          </div>
+          <div className="form-group">
+            <button type="submit" onClick={() => { this.addUser() }} className="btn btn-primary btn-block"> Create Account  </button>
```
Comment (ibrahimseckin) [inline client/src/components/login/components/registration/registration.js:57]: I will add localization

File: client/src/components/login/components/registration/registration.js [hunk 6]
```diff
@@ -2,33 +2,62 @@ import React, { Component } from 'react'
 import './registration.css'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faUser } from '@fortawesome/free-solid-svg-icons';
+import axios from 'axios';
+import { API_URL } from '../loginPanel/config'
```
Comment (ibrahimseckin) [inline client/src/components/login/components/registration/registration.js:6]: can we change this api url variable name?

File: backend/app/controllers/loginController.js (hunk 1 above)
Comment (erolcanozgen) [reply backend/app/controllers/loginController.js:12]: the result of this method will be used after login successful. User information will be used in home page

File: backend/server.js (hunk 4 above)
Comment (erolcanozgen) [reply backend/server.js:51]: server.js is delegating the routing of the login operation to the routes.loginRoutes

File: client/src/components/login/components/registration/registration.js (hunk 6 above)
Comment (erolcanozgen) [reply client/src/components/login/components/registration/registration.js:6]: what would be your proposal ?

File: backend/lib/auth.controller.js (hunk 3 above)
Comment (erolcanozgen) [reply backend/lib/auth.controller.js:3]: I will remove it

File: client/src/components/login/components/registration/registration.js (hunk 6 above)
Comment (ibrahimseckin) [reply client/src/components/login/components/registration/registration.js:6]: authentication_url?

**Pull request merged**